libqtile/core/lifecycle.py

```diff
@@ -19,6 +19,7 @@ class LifeCycle:
     # referenced here when atexit fires will NOT be finalized properly.
     def __init__(self) -> None:
         self.behavior = Behavior.NONE
+        self.exitcode: int = 0
```

Comment:
This seems unused? Also, can you add a test?

Reply:
Oh, no, it is used :). But can you add a test?

Reply:
It is not stricly needed to specify the member here, because python will just add the variable dynamically (from stop method). This is used to save the exitcode between termination of the shutdown command method and the real shutdown in LifeCycle class. Of course the member variable is only needed for a rather short amount of time. But I did not see another way to communicate the exitcode to "the last line" of qtile in its execution.
Should I try to write a test starting qtile, issuing the shutdown command and checking the exitcode?

Reply:
> Should I try to write a test starting qtile, issuing the shutdown command and checking the exitcode?

Yep, that sounds great, thank you.

>  But I did not see another way to communicate the exitcode to "the last line" of qtile in its execution.

Yep, no worries, it's fine as is, I was just confused.